mutex.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package mutex
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/juju/errors"
+)
+
+var (
+	validName = regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9-]*$")
+)
+
+// Releaser defines the Release method that is the only thing that can be done
+// to a acquired mutex.
+type Releaser interface {
+	// Release makes the mutex available for others and invalidates the instance.
+	// Subsequent calls to Release will error.
+	Release() error
```

Comment:
Calling Release twice should panic for the same reason that closing a channel twice panics. This is a programming error, and laziness begats laziness.